fe1-web/model/network/method/message/data/MessageRegistry.ts

```diff
@@ -80,9 +115,45 @@ export class MessageRegistry {
     }
     const { handle } = messageEntry;
     if (handle === undefined) {
-      console.warn(`Message ${data.object} ${data.action} does not have a 'handle' property in MessageRegistry`);
+      console.warn(`Message ${data.object} ${data.action} does not have an 'handle' property in MessageRegistry`);
       return false;
     }
     return handle(msg);
   }
+
+  /**
+   * Builds a message from a MessageData by calling the corresponding function.
+   *
+   * @param data -The type of message to be built
+   * @returns MessageData - The built message
+   */
+  buildMessageData(data: MessageData): MessageData {
+    const messageEntry = this.mapping.get(k(data.object, data.action));
+    if (messageEntry === undefined) {
+      throw new Error(`Message ${data.object} ${data.action} is not contained in MessageRegistry`);
+    }
+    const { build } = messageEntry;
+    if (build === undefined) {
+      throw new Error(`Message ${data.object} ${data.action} does not have an 'build' property in MessageRegistry`);
+    }
+    return build(data);
+  }
+
+  /**
+   * Gets the signature type of a message data.
+   *
+   * @param data - The message data we want to know how to sign
+   * @returns SignatureType - The type of signature of the message
+   */
+  getSignatureType(data: MessageData): SignatureType {
+    const messageEntry = this.mapping.get(k(data.object, data.action));
+    if (messageEntry === undefined) {
+      throw new Error(`Message ${data.object} ${data.action} is not contained in MessageRegistry`);
+    }
+    const { signature } = messageEntry;
+    if (signature === undefined) {
+      throw new Error(`Message ${data.object} ${data.action} does not have a 'signature' property in MessageRegistry`);
+    }
+    return signature;
```

Comment:
as we discussed, we can probably have a single verification at the application startup, to fail fast / fail hard